packages/network-controller/tests/NetworkController.test.ts

```diff
@@ -3616,6 +3622,204 @@ describe('NetworkController', () => {
       });
     });
   });
+
+  describe('rollbackToPreviousProvider', () => {
```

Comment:
There are some other tests that I've added on the extension side to ensure that all of the logic that happens when calling `setProviderType` or `setActiveNetwork` (well, `setRpcTarget`, but same thing) happens in this case. Can we try copying over those tests and then tweaking them to work with this version of the network controller? I'm trying to get these unit tests to be as similar as possible to make comparing the two versions easier as we bring them closer and closer together.

Reply:
added roughly all the tests that apply from the extension side!